core/src/Revolution/modTemplateVar.php

```diff
@@ -984,13 +993,30 @@ public function processBindings($value = '', $resourceId = 0, $preProcess = true
     public function parseBinding($binding_string)
     {
         $match = [];
+        $match2 = [];
         $binding_string = trim($binding_string);
         $regexp = '/@(' . implode('|', $this->bindings) . ')\s*(.*)/is'; /* Split binding on whitespace */
+        
         if (preg_match($regexp, $binding_string, $match)) {
             /* We can't return the match array directly because the first element is the whole string */
+            
+            $regexp2 = '/(\S+)\s+(.+)/is'; /* Split binding on second whitespace to get properties */
+            
+            $properties = [];
+            if (preg_match($regexp2, $match[2] , $match2)) {
+                if (isset($match2[2])) {
+                    $props = json_decode($match2[2],true);
+                    if (is_array($props)){
```

Comment:
Maybe it is useful to check for json_last_error instead and log the issue somehow.

See this method from https://stackoverflow.com/questions/6041741/fastest-way-to-check-if-a-string-is-json-in-php

```
function json_validate($string)
{
    // decode the JSON data
    $result = json_decode($string);

    // switch and check possible JSON errors
    switch (json_last_error()) {
        case JSON_ERROR_NONE:
            $error = ''; // JSON is valid // No error has occurred
            break;
        case JSON_ERROR_DEPTH:
            $error = 'The maximum stack depth has been exceeded.';
            break;
        case JSON_ERROR_STATE_MISMATCH:
            $error = 'Invalid or malformed JSON.';
            break;
        case JSON_ERROR_CTRL_CHAR:
            $error = 'Control character error, possibly incorrectly encoded.';
            break;
        case JSON_ERROR_SYNTAX:
            $error = 'Syntax error, malformed JSON.';
            break;
        // PHP >= 5.3.3
        case JSON_ERROR_UTF8:
            $error = 'Malformed UTF-8 characters, possibly incorrectly encoded.';
            break;
        // PHP >= 5.5.0
        case JSON_ERROR_RECURSION:
            $error = 'One or more recursive references in the value to be encoded.';
            break;
        // PHP >= 5.5.0
        case JSON_ERROR_INF_OR_NAN:
            $error = 'One or more NAN or INF values in the value to be encoded.';
            break;
        case JSON_ERROR_UNSUPPORTED_TYPE:
            $error = 'A value of a type that cannot be encoded was given.';
            break;
        default:
            $error = 'Unknown JSON error occured.';
            break;
    }

    if ($error !== '') {
        // throw the Exception or exit // or whatever :)
        exit($error);
    }

    // everything is OK
    return $result;
}
```

Reply:
Such a validation method can go into in a global (helper) class.

Reply:
yeah, testing and logging all possible mistakes. like valid json or not existing snippets, chunks...  could be an additional feature request

Reply:
Until then you can check for json_last_error and log a debug message with 'invalid JSON'.

Reply:
check added